src/main/scala/chisel3/util/BlackBoxUtils.scala

```diff
@@ -19,7 +38,11 @@ trait HasBlackBoxResource extends BlackBox {
     */
   def addResource(blackBoxResource: String): Unit = {
     val anno = new ChiselAnnotation with RunFirrtlTransform {
-      def toFirrtl = BlackBoxResourceAnno(self.toNamed, blackBoxResource)
+      def toFirrtl = {
+        ResourceHelpers.getResourceAsString(blackBoxResource) match {
+          case ResourceHelpers.ResourceResult(fileName, contents) => BlackBoxInlineAnno(self.toNamed, fileName, contents)
```

Comment:
```
        val blackboxFile = os.resource / blackBoxResource
        BlackBoxInlineAnno(self.toNamed, blackboxFile.baseName, os.read(blackboxFile))
```
Have a try to type-safe way to read from file:)

Reply:
Switched to `os-lib`. :+1: